GVFS/GVFS/CommandLine/IncludeVerb.cs

```diff
@@ -0,0 +1,182 @@
+﻿using CommandLine;
+using GVFS.Common;
+using GVFS.Common.Database;
+using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace GVFS.CommandLine
+{
+    [Verb(IncludeVerb.IncludeVerbName, HelpText = "List, add, or Remove from the list of folder that are included to project")]
+    public class IncludeVerb : GVFSVerb.ForExistingEnlistment
+    {
+        private const string IncludeVerbName = "include";
+
+        [Option(
+            'a',
+            "add",
+            Required = false,
+            Default = "",
+            HelpText = "A semicolon-delimited list of folders to include. Wildcards are not supported.")]
+        public string Add { get; set; }
+
+        [Option(
+            'r',
+            "remove",
+            Required = false,
+            Default = "",
+            HelpText = "A semicolon-delimited list of folders to remove for being included. Wildcards are not supported.")]
+        public string Remove { get; set; }
+
+        [Option(
+            'l',
+            "list",
+            Required = false,
+            Default = false,
+            HelpText = "List of folders to for being included in projection.")]
+        public bool List { get; set; }
+
+        protected override string VerbName => IncludeVerbName;
+
+        protected override void Execute(GVFSEnlistment enlistment)
+        {
+            using (JsonTracer tracer = new JsonTracer(GVFSConstants.GVFSEtwProviderName, "Include"))
+            {
+                try
+                {
+                    tracer.AddLogFileEventListener(
+                        GVFSEnlistment.GetNewGVFSLogFileName(enlistment.GVFSLogsRoot, GVFSConstants.LogFileTypes.Include),
+                        EventLevel.Informational,

```

Comment:
I'm questioning the value of even having the logs for this verb specifically or if we should just output to the console any error that happen?  Otherwise there could be a lot of include logs that will have one line or that status failed.  I guess there might be other errors that would be good to know about but I think for this verb it would be nice if we had just one file instead of a file for every invocation of the verb. 

Most of my logs only have this.
```
[2019-07-16 16:41:26 -06:00] Include (Stop) {"DurationMs":14044}
```

What are your thoughts?

Reply:
> I'm questioning the value of even having the logs for this verb specifically or if we should just output to the console any error that happen?

> Otherwise there could be a lot of include logs that will have one line or that status failed.

> Most of my logs only have this.

Maybe we should only create a log file when users specify folders to add or remove?  I would guess we won't get much value from having logs when `gvfs include` was only used to list the current contents.  However, I could see it being useful to see the history of how folders were added/removed when investigating support issues.

If we're not already I think it would be helpful to log the arguments to the verb so we know what is being added or removed.

What do you think?

Reply:
Sounds good thanks!